### DISCUSSION TITLE
Multiple GPUs with ONNX and 0.15-beta3

### DISCUSSION THREAD

**Discussion by jvallery** [blakeblackshear/frigate#15591] [Ask A Question]
I'm trying to configure 0.15-beta3 to work with multiple GPUs and the onnx detector. I've tried this a few ways, but had no luck so far.

If I specify multiple detectors without a specific device ID, 2 instances are created but they are both running on gpu 0. If I specify the "device" property in the config with the integer associated with the GPU, the config fails to load as invalid.

How do I specify which GPU each detector should run on?

`
detectors:
  onnx_0:
    type: onnx
  onnx_1:
    type: onnx
`
  51732 root   0  Compute  13%    480MiB   4%    26%    863MiB frigate.detector.onnx_1
  51715 root   0  Compute  25%    404MiB   3%    28%    860MiB frigate.detector.onnx_0

`+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.183.01             Driver Version: 535.183.01   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 3060        Off | 00000000:21:00.0 Off |                  N/A |
| 58%   61C    P2              89W / 170W |   9454MiB / 12288MiB |     91%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce RTX 3060        Off | 00000000:2B:00.0 Off |                  N/A |
|  0%   49C    P8              18W / 170W |      3MiB / 12288MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
+---------------------------------------------------------------------------------------+`

**Comment by NickM-27** [comment]
You need to specify the device as a string not an integer

**Comment by jvallery** [comment]
Yes! That worked. Thanks.

Not obvious from the docs. The tensorrt example uses an integer.
